pkg/pool-hooks/contracts/StableSurgeHook.sol

```diff
@@ -16,286 +16,100 @@ import { StableMath } from "@balancer-labs/v3-solidity-utils/contracts/math/Stab
-        uint256[] memory amountsOutScaled18,
-        uint256[] memory amountsOutRaw,
-        uint256[] memory balancesScaled18,
-        bytes memory
-    ) public view override returns (bool success, uint256[] memory hookAdjustedAmountsOutRaw) {
-        // Proportional remove is always fine.
-        if (kind == RemoveLiquidityKind.PROPORTIONAL) {
-            return (true, amountsOutRaw);
-        }
-
-        // Rebuild old balances before removing liquidity.
-        uint256[] memory oldBalancesScaled18 = new uint256[](balancesScaled18.length);
-        for (uint256 i = 0; i < balancesScaled18.length; ++i) {
-            oldBalancesScaled18[i] = balancesScaled18[i] + amountsOutScaled18[i];
-        }
-
+        uint256[] memory oldBalancesScaled18,
+        uint256[] memory balancesScaled18
+    ) internal view override returns (bool isSurging) {
         SurgeFeeData memory surgeFeeData = _surgeFeePoolData[pool];
-        uint256 newTotalImbalance = StableSurgeMedianMath.calculateImbalance(balancesScaled18);
 
-        bool isSurging = _isSurging(surgeFeeData, oldBalancesScaled18, newTotalImbalance);
+        uint256 oldTotalImbalance = StableSurgeMedianMath.calculateImbalance(oldBalancesScaled18);
+        uint256 newTotalImbalance = StableSurgeMedianMath.calculateImbalance(balancesScaled18);
 
-        // If we're not surging, it's fine to proceed; otherwise halt execution by returning false.
-        return (isSurging == false, amountsOutRaw);
+        isSurging = _isSurging(surgeFeeData.thresholdPercentage, oldTotalImbalance, newTotalImbalance);
     }
 
-    /// @inheritdoc IStableSurgeHook
-    function setMaxSurgeFeePercentage(
-        address pool,
-        uint256 newMaxSurgeSurgeFeePercentage
-    ) external withValidPercentage(newMaxSurgeSurgeFeePercentage) onlySwapFeeManagerOrGovernance(pool) {
-        _setMaxSurgeFeePercentage(pool, newMaxSurgeSurgeFeePercentage);
-    }
+    /***************************************************************************
+                                  Legacy Functions
+    ***************************************************************************/
 
-    /// @inheritdoc IStableSurgeHook
-    function setSurgeThresholdPercentage(
-        address pool,
-        uint256 newSurgeThresholdPercentage
-    ) external withValidPercentage(newSurgeThresholdPercentage) onlySwapFeeManagerOrGovernance(pool) {
-        _setSurgeThresholdPercentage(pool, newSurgeThresholdPercentage);
-    }
-
-    /**
-     * @notice Calculate the surge fee percentage. If below threshold, return the standard static swap fee percentage.
-     * @dev It is public to allow it to be called off-chain.
-     * @param params Input parameters for the swap (balances needed)
-     * @param pool The pool we are computing the fee for
-     * @param staticFeePercentage The static fee percentage for the pool (default if there is no surge)
-     */
-    function _getSurgeFeePercentage(
+    function getSurgeFeePercentage(
```

Comment:
I kept this function to be compatible with old versions of Stable Surge Hook. There's a new function called computeSwapSurgeFeePercentage that does exactly the same thing and is located in the SurgeHookCommon contract, so I'm not 100% sure we need to keep this one.